### PR TITLE
Time and Space now triggers correctly for twin dice

### LINKED ISSUE #1593: Time & Space (1,Y=2) die rolled odd but didn't give another turn
In http://dev.buttonweavers.com/ui/game.html?game=1599:

    irilyth performed Skill attack using [gn(X=20):10,`^(1,Y=2):3] against [gn(X=4):4]; Defender gn(X=4) was captured; Attacker gn(X=20) rerolled 10 => 19; Attacker `^(1,Y=2) recipe changed from `^(1,Y=2) to ^(1,Y=2), rerolled 3 => 3

But I didn't get another turn. Screenshot:

bm

### PR DESCRIPTION
**Pull request by blackshadowshade**
Fixes #1593. Also addresses the action logging wording left over from comments on #1583.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/678/

### CODE REVIEW

File: test/src/api/responderTest.php [hunk 1]
```diff
@@ -9741,4 +9747,127 @@ public function test_interface_game_030() {
...
+        // responder003's idle ornery dice rerolled at end of turn: Ho(W=4)? changed size from 4 to 12 sides, recipe changed from Ho(W=4)? to Ho(W=12)?, rerolled 2 => 8
+
+        $this->verify_api_submitTurn(
+            array(1, 2, 3, 8),
+            'responder003 performed Power attack using [^B(T=3,T=3):4] against [(20):4]; Defender (20) was captured; Attacker ^B(T=3,T=3) rerolled 4 => 3. responder003 gets another turn because a Time and Space die rolled odd. responder003\'s idle ornery dice rerolled at end of turn: Ho(W=4)? changed size from 4 to 12 sides, recipe changed from Ho(W=4)? to Ho(W=12)?, rerolled 2 => 8. ',
+            $retval, array(array(0, 3), array(1, 0)),
+            $gameId, 1, 'Power', 0, 1, '');
+
+        $this->update_expected_data_after_normal_attack(
+            $expData, 1, array('Power', 'Skill'),
```
Comment (cgolubi1) [inline test/src/api/responderTest.php]: Hmm, i'm suspicious --- the expected data update shorthand probably obscures this, and i should have stuck the word BUG in a few more places (for both of which pieces of unclarity i apologise), but i think the first argument is "what should the value of `$activePlayerIdx` be after this attack?"  And it should still be 0.  So i suspect we got the game action log message, but the fix didn't actually work.  Let me try this in a sandbox and get you a screencap either way.

**Comment by cgolubi1** [comment]
Huh, nope, your fix does work.  I wonder what i'm looking at...

**Comment by cgolubi1** [comment]
Ran some regression test games, and they look fine too.  Merging.

File: test/src/api/responderTest.php (hunk 1 above)
Comment (blackshadowshade) [inline test/src/api/responderTest.php]: Ah, I must have forgotten that. I've obviously manually changed these attributes afterwards (see lines 9864-9866) to get the test to pass. I'm happy for you to update the test to match the correct syntax.

**Comment by jimmosk** [comment]
Actually, I think I'm still encountering this bug. In game 1614 this happened:

(2015-04-27 20:49:22)	jimmosk performed Power attack using [^B(T=3,T=3):3] against [p(4):3]; Defender p(4) was captured; Attacker ^B(T=3,T=3) rerolled 3 => 3.  
But then my opponent got to go.

Except, wait, at that point I was left with no legal attacks. So maybe it did give me an extra turn, and the actual bug was that it didn't display that I had gotten an extra turn nor that I had autopassed... I can't tell.

**Comment by irilyth** [comment]
Looking at the log, I think your opponent got a turn before you you passed, and that you didn't get an extra turn. (Also, if you do get an extra turn, but then have to pass, I'm pretty sure that gets logged -- I think I've seen it.)

I think this fix has been merged to buttonmen-dev:master, but may not have been deployed to the site yet.

**Comment by cgolubi1** [comment]
Ah, you're right.  I'll do a deployment to dev tonight.

**Pull request merged**